target/linux/ath79/dts/qca9531_comfast_cf-wr752ac.dts

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca953x.dtsi"
+
+/{
+    compatible = "comfast,cf-wr752ac", "qca,qca9531";
+	model = "COMFAST CF-WR752AC";
+    
+    chosen {
+        bootargs = "console=ttyS0,115200";
+    };
+
+    leds{
+        wlan5g {
+			label = "cf-wr752:green:wlan5g";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+        wlan2g {
+			label = "cf-wr752:blue:wlan2g";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1tpt";
+		};
+        status {
+			label = "cf-wr752:red:lan";
+			gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+		};
+    };
+
+    keys {
+		compatible = "gpio-keys";
+        
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+    };
+};
+
+&pcie0 {
+	status = "okay";
+};
+
+&uart {
+	status = "okay";
+};
+
+&gpio {
+	status = "okay";
+};
+
+&spi {
+	    status = "okay";
+	    num-cs = <1>;
+
+    flash@0 {
+	    compatible = "jedec,spi-nor";
+	    reg = <0>;
+	    spi-max-frequency = <25000000>;
+
+	    partitions {
+		    compatible = "fixed-partitions";
+		    #address-cells = <1>;
+	        #size-cells = <1>;
+
+		    partition@0 {
+			    label = "u-boot";
+			    reg = <0x0 0x10000>;
+			    read-only;
+		    };
+
+            art: partition@10000 {
+			    label = "art";
+			    reg = <0x10000 0x10000>;
+			    read-only;
+            };
+
+            partition@20000 {
+                    label = "firmware";
+                    reg = <0x20000 0xfc0000>;
+					compatible = "denx,uimage";         
+            };
+
+			partition@fe0000 {
+                    label = "configs";
+                    reg = <0x0fe0000 0x10000>;
+            };
+
+            partition@ff0000 {
+            	label = "nvram";    
+                reg = <0x0ff0000 0x10000>;
+            };
+
+        };
+	};
+};
+
+&wmac {
+    status = "okay";
+    mtd-cal-data = <&art 0x1000>;
+	mtd-mac-address = <&art 0x6>;
```

Comment:
Since you currently just randomly assign MAC address, please check which interface has which addresses and report them back as explained here:
https://openwrt.org/docs/guide-developer/mac.address

Reply:
I've got the following mac address to appear on my art partition.
mtd1_art @ 0x10000 :
0x0		xx:xx:xx:xx:dc:f8
0x6		xx:xx:xx:xx:dc:fa
0x1002		xx:xx:xx:xx:dc:f9
0x5006		xx:xx:xx:xx:dc:fb

I tought of setting up like the following:
LAN = xx:xx:xx:xx:dc:f8
WiFi 2.4G = xx:xx:xx:xx:dc:fa
WiFi 5G = xx:xx:xx:xx:dc:fb

This device only has one ethernet port so I 've set up as LAN port.
I believe I still need to make more changes to the code as I booted my device and got the following:
LAN             xx:xx:xx:xx:dc:f9
Wifi 2.4G     xx:xx:xx:xx:dc:fb
Wifi 5G        xx:xx:xx:xx:dc:fa

For reference the original firmware had the folloing definitions:
 xx:xx:xx:xx:dc:f8  lan port
xx:xx:xx:xx:dc:fa wifi 5GHz
xx:xx:xx:xx:dd:02 wifi 2.4GHz

I'll try some code changes and report back

Reply:
Your stock firmware values are strange, particularly since you have valid-looking MAC addresses in the caldata but they are not used at all. Are you sure about those?

Nevertheless, at least the lan MAC address matches 0x0, so please use this one for eth0. Until we are sure about the real values for the WiFi interfaces, we might as well use the addresses from caldata (0x1002 and 0x5006), which means that you would remove anything _overwriting_ the MAC address for them (i.e. ath10k_patch_mac and mtd_mac_address). Then we should end up with the following interim assignment:
lan :f8
2.4 GHz :f9
5 GHz :fb